src/main/java/pl/zalwi/data/UserRepository.java

```diff
@@ -0,0 +1,73 @@
+package pl.zalwi.data;
+
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Repository
+public class UserRepository {
+
+    private List<User> userList;
+
+    public UserRepository(){
+        userList = new ArrayList<User>();
+        userList.add(new User("Kamil", "Zalwert", 31));
+        userList.add(new User("Kamil", "Drugi", 32));
+        userList.add(new User("Jan", "Kowalski", 21));
+        userList.add(new User("John", "Kowalski", 45));
+        userList.add(new User("Piotr", "Pierwszy", 11));
+        userList.add(new User("Mietek", "Jedenastolate", 11));
+    }
+
+    public List<User> getAll(){
+        return userList;
+    }
+
+    public void addUser(User user){
+        userList.add(user);
+    }
+    public void delUsers(List<User> userToDelete){
+        userList.removeAll(userToDelete);
+    }
+
+    public List<User> findUsers(String firstName, String lastName, Integer age){
+        List<User> resultList = new ArrayList<User>();
+        int inputsBitFlag=0;
+
+        if(!firstName.isEmpty()) inputsBitFlag+=1;
```

Comment:
Zauważyłem, że bardzo lubisz opierać się na "magicznych" liczbach które mają jakieś specialne znaczenie. Generalnie w Javie unikamy takiego podejścia bo zaciemnia ono czytanie kodu a co gorsza ciężej je się modyfikuje. Zauważ że to co my chcieliśmy tutaj osiągnąć to sprawdzenie czy dany użytkownik pasuje do wpisanych "filtrów". Wiemy że każdy parametr jest opcjonalny, więc możemy założyć że jeżeli ktoś czegoś nie wpisał to nie musimy nakładać kolejnych "obostrzeń" jeśli chodzi o użytkowników. Tak więc każdy użytkownik z danym atrybutem spełnia oczekiwania kogoś kto z tego korzysta. Wystarczy się więc po wszystkich użytkownikach przeiterować i sprawdzić czy dany użytkownik ma takie same atrybuty które zostały podane przez kogoś korzystającego z naszego systemu. Sprawdzamy więc dla każdego użytkownika czy dany atrybut został wpisany (czy coś jest null bądź puste) oraz czy pasuje on do tego co ma w sobie użytkownik. I tak dla każðego atrybutu podanego przez użytkownika. W kodzie wyglądałoby to następująco:
```
private boolean checkSearchConditionsForUser(String imie, String nazwisko, Integer wiek, User user) {
        boolean nameOk = true;
        boolean surnameOk = true;
        boolean ageOk = true;

        if(!StringUtils.isEmpty(imie) && !imie.equals(user.getName())) {
            nameOk = false;
        }

        if(!StringUtils.isEmpty(nazwisko) && !nazwisko.equals(user.getLastName())) {
            surnameOk = false;
        }

        if(wiek != null && wiek != user.getAge()) {
            ageOk = false;
        }

        return nameOk && surnameOk && ageOk;
    }

```